presto-spi/src/main/java/com/facebook/presto/spi/statistics/HistoricalStatisticsStorageService.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.statistics;
+
+import java.util.List;
+import java.util.Map;
+
+public interface HistoricalStatisticsStorageService
```

Comment:
Is it possible to merge this with `ExternalPlanStatisticsProvider`? We can call it `ExternalPlanStatisticsManager` or whatever you feel reasonable. (Because I find the name to be confusing with `HistoryBasedPlanStatisticsManager`). 

The reason is that both the provider and storage service are separate plugins. But I don't see a case that a user would plug two different stats infras with one on "read" and one on "write". They are always combined. So logically, it's just one external plugin.

Reply:
Will try implementing something along those lines.